src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -98,7 +111,23 @@ protected override ParameterBuilder DefineParameterCore(int position, ParameterA
             return parameter;
         }
 
-        protected override ILGenerator GetILGeneratorCore(int size) => throw new NotImplementedException();
+        protected override ILGenerator GetILGeneratorCore(int size)
+        {
+            if (IsGenericMethod && !IsGenericMethodDefinition)
+            {
+                throw new InvalidOperationException();
+            }
+
+            if ((_methodImplFlags & MethodImplAttributes.CodeTypeMask) != MethodImplAttributes.IL ||
+                (_methodImplFlags & MethodImplAttributes.Unmanaged) != 0 ||
+                (_attributes & MethodAttributes.PinvokeImpl) != 0)
```

Comment:
The CoreLib implementation also has a check for DllImport here. Is it intentionally omitted?

Reply:
We are setting the attribute when `DllImportAttribute` is set, no need to keep another boolean flag for this https://github.com/dotnet/runtime/blob/42a6776e6cb1569fe277181f424ccbbe243cf386/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs#L111-L114